### DISCUSSION TITLE
Customize Management Plugin in Kubernetes Cluster

### DISCUSSION THREAD

**Discussion by NuwanSameera** [rabbitmq/rabbitmq-server#5275] [Other] [answered]
I added new API to management plugin to `delete_mqtt_connection_by_username`. I need to delete all connections related to given username. All the changes are working properly in local machine and inside docker image. But after deploying this docker image in Kubernetes cluster. It has three nodes and management plugin enabled. 

Management plugin port 15672 port through LoadBalancer ingress. 

When I sending DELETE request to PUBLIC_IP:15672 hit to only one pod. If that pod not include requested connection return not found.  

I think problem in here [1].

[1]. https://github.com/NuwanSameera/rabbitmq-server/blob/5407bf22a6c4854d93133fe64ad784e36994c88b/deps/rabbit/src/rabbit_connection_tracking.erl#L357

How I execute this DB query on all the nodes ?

**Comment by lukebakken** [comment, marked as answer]
You need to modify your code so that all nodes are queried. See the following function:

https://github.com/NuwanSameera/rabbitmq-server/blob/feature-delete-connection-by-name/deps/rabbit/src/rabbit_connection_tracking.erl#L340-L347

A big hint is that the `Nodes` variable is not used in your `lookup_by_user` function.

**Reply by NuwanSameera** [reply]
Thank you very much. Working after adding scan data with all running nodes.
